runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -202,6 +203,18 @@ std::unique_ptr<onert::ir::Model> loadModel(const std::string filename,
   return std::unique_ptr<onert::ir::Model>(nullptr);
 }
 
+std::unique_ptr<onert::ir::train::TrainingInfo>
+loadTrainingInfo(const std::shared_ptr<onert::ir::Model> &model)
+{
+  const auto tinfo_name = onert::train::traininfo_loader::TRAININFO_METADATA_NAME;
+  if (model->exists_metadata(tinfo_name))
+  {
+    const auto buffer = model->extract_metadata(tinfo_name);
```

Comment:
I have a question regardless to this PR.
Is `model` going to always have `metadata` as external data(i.g. `ir::MMapedData`) in the future?

Reply:
I'll ask the above question in another issue since this PR is closed.

Reply:
Let me clarify the word first, to make sure that we have the same understanding. 

* `ir::CachedData` : holds **copy** of original data
* `ir::ExternalData` : points original data, we have to make sure that the original data is not lost.

Reply:
> Is model going to have metadata as only external data(i.g. ir::MMapedData) in the future? 

@ragmani 
For now, only `TrainingInfo` is passed through Model's Metadata.
Since TrainingInfo is extracted when **loaded** time, we could use `ExternalData`.
If Metadata is used(extracted) at another time, I'm not sure that the original buffer (mmaped data) is not damaged. 

So, About your question, For now, Yes only external data is used in metadata.
About the future, I'm not 100% sure.

Reply:
This PR was closed. So I copied the above comment into #12433